webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java

```diff
@@ -0,0 +1,114 @@
+ *  Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ ******************************************************************************/
+package org.sakaiproject.webapi.controllers;
+
+import org.sakaiproject.assignment.api.AssignmentService;
+import org.sakaiproject.assignment.api.model.Assignment;
+import org.sakaiproject.assignment.api.model.AssignmentSubmission;
+import org.sakaiproject.entity.api.EntityManager;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.exception.PermissionException;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.user.api.UserDirectoryService;
+import org.sakaiproject.webapi.beans.AssignmentRestBean;
+import org.sakaiproject.webapi.beans.AssignmentSubmissionRestBean;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ */
+@Slf4j
+@RestController
+public class AssignmentsController extends AbstractSakaiApiController {
+
+	@Autowired
+    SiteService siteService;
+
+    @Autowired
+    AssignmentService assignmentService;
+
+    @Autowired
+    EntityManager entityManager;
+
+    @Autowired
+    UserDirectoryService userDirectoryService;
+
+    @GetMapping(value = "/sites/{siteId}/assignments", produces = MediaType.APPLICATION_JSON_VALUE)
+    public List<AssignmentRestBean> getSiteAssignments(@PathVariable String siteId) {
+
+        checkSakaiSession();
+        List<AssignmentRestBean> beanList = assignmentService.getAssignmentsForContext(siteId).stream().map(assignment -> {
+            AssignmentRestBean bean = new AssignmentRestBean(assignment);
+            bean.setStatus(assignmentService.getAssignmentStatus(assignment.getId()));
+            bean.setScale(assignmentService.getScaleDisplay(assignment.getId()));
+            bean.setAccess(assignmentService.getAccessDisplay(assignment.getId()));
+            String assignmentReference = assignmentService.assignmentReference(siteId, assignment.getId());
+            bean.setTotalSubmissions(assignmentService.countSubmissions(assignmentReference, null));
+            bean.setNewSubmissions(assignmentService.countSubmissions(assignmentReference, false));
+            return bean;
+        }).collect(Collectors.toList());
+        return beanList;
+    }
+
+    @GetMapping(value = "/assignments/{assignmentId}/submissions", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
I think you should stick to /sites/{siteId}/assignments/{assignmentId}/submissions. I know it's a bit longer, but it's consistent with other parts of the webapi. But, I'm open to discussion.